calico_node/startup/startup.go

```diff
@@ -660,18 +695,27 @@ func createIPPool(client *client.Client, cidr *net.IPNet, ipipModeName string, i
 
 // checkConflictingNodes checks whether any other nodes have been configured
 // with the same IP addresses.
-func checkConflictingNodes(client *client.Client, node *api.Node) {
+func checkConflictingNodes(ctx context.Context, client client.Interface, node *api.Node) {
 	// Get the full set of nodes.
 	var nodes []api.Node
-	if nodeList, err := client.Nodes().List(api.NodeMetadata{}); err != nil {
+	if nodeList, err := client.Nodes().List(ctx, options.ListOptions{}); err != nil {
 		fatal("Unable to query node confguration: %s", err)
 		terminate()
 	} else {
 		nodes = nodeList.Items
 	}
 
-	ourIPv4 := node.Spec.BGP.IPv4Address
-	ourIPv6 := node.Spec.BGP.IPv6Address
+	ourIPv4, _, err := cnet.ParseCIDR(node.Spec.BGP.IPv4Address)
```

Comment:
I thought I'd mentioned this,  I think we should allow just an IP address to be specified too.  So I think you should use the ParseCIDROrIP function - that'll also parse an IP and assume full mask for the network.